### PR TITLE
KH2FM: Rendering updates and map name fixes

### PR DESCRIPTION
**Pull request by Murugo**
- Remove explicit camera clip planes in the renderer.
- Add support for fake reflections for some textures such as the glass doors in Twilight Town Station Plaza.
- Update names for some maps to be more consistent with the the list here: https://openkh.dev/kh2/worlds.html

### CODE REVIEW

File: src/kh2fm/render.ts [hunk 1]
```diff
@@ -419,6 +430,9 @@ export class MapData {
                 vBuffer[vBufferIndex++] = texScaleOffs[3];
                 vBuffer[vBufferIndex++] = mesh.uvScroll[0];
                 vBuffer[vBufferIndex++] = mesh.uvScroll[1];
+                vBuffer[vBufferIndex++] = mesh.normal.length > 0 ? mesh.normal[i][0] : 0;
+                vBuffer[vBufferIndex++] = mesh.normal.length > 0 ? -mesh.normal[i][1] : 0;
+                vBuffer[vBufferIndex++] = mesh.normal.length > 0 ? mesh.normal[i][2] : 0;
```
[the remark this answers is not in the document]
Comment (Murugo) [reply src/kh2fm/render.ts:435]: This was an attempt to flip the resulting V texture coordinate outside of the shader, but I don't believe this was correct. I made a correction to the shader instead.

File: src/kh2fm/render.ts [hunk 2]
```diff
@@ -379,6 +383,13 @@ export class MapData {
                     -texture.clipLeft / texture.textureAnim.sheetWidth,
                     -texture.clipTop / texture.textureAnim.sheetHeight
                 );
+            } else if (mesh.useNormal) {
+                vec4.set(texScaleOffs,
+                    mesh.texture!.width() / this.atlasWidth * 0.5,
+                    mesh.texture!.height() / this.atlasHeight * 0.5,
+                    ((mesh.textureBlock.atlasX + mesh.texture!.clipLeft) * 2 + mesh.texture!.width()) / this.atlasWidth * 0.5,
+                    ((mesh.textureBlock.atlasY + mesh.texture!.clipTop) * 2 + mesh.texture!.height()) / this.atlasHeight * 0.5,
+                );
```
[the remark this answers is not in the document]
Comment (Murugo) [reply src/kh2fm/render.ts:392]: Right, I think there is an opportunity to simplify this. I will look into cleaning this up in the near future.

File: src/kh2fm/program.glsl [hunk 3]
```diff
@@ -46,11 +49,16 @@ void main() {
     vec4 t_Color = vec4(0.5, 0.5, 0.5, 1);
 
 #ifdef USE_TEXTURE
+#ifdef USE_NORMAL
+    vec2 tc = (mat3(u_View[0], u_View[1], u_View[2]) * v_Normal).xy;
```
[the remark this answers is not in the document]
Comment (Murugo) [reply src/kh2fm/program.glsl]: Ah thanks, this is an interesting case I was not aware of. Applied the fix here.

File: src/kh2fm/render.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (magcius) [reply src/kh2fm/render.ts:392]: No rush.

**Pull request merged**